File: src/main/java/org/example/easy/A090MovingAVGFromDStream/A090Exercise.java
package org.example.easy.A090MovingAVGFromDStream;

import java.util.LinkedList;
import java.util.Queue;

public class A090Exercise {
    private Queue<Integer> queue;
    private double sum;
    private final int size;

    public A090Exercise(int size) {
        this.queue = new LinkedList<>();
        this.sum = 0.0;
        this.size = size;
    }

    public double next(int val) {
        if (queue.size() == size) {
            sum -= queue.poll();
        }
        queue.offer(val);
        sum += val;
        return sum / queue.size();
    }

    public static void main(String[] args) {
        A090Exercise m = new A090Exercise(3);
        System.out.println(m.next(1));
        System.out.println(m.next(10));
        System.out.println(m.next(3));
        System.out.println(m.next(5));
    }
}

File: src/main/java/org/example/easy/A090MovingAVGFromDStream/A090Exercise.kt
import java.util.*

class Exercise(private val size: Int) {
    private val queue: Queue<Int> = LinkedList()
    private var sum = 0.0

    fun next(`val`: Int): Double {
        if (queue.size == size) {
            sum -= queue.poll().toDouble()
        }
        queue.offer(`val`)
        sum += `val`.toDouble()
        return sum / queue.size
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val m = Exercise(3)
            println(m.next(1))
            println(m.next(10))
            println(m.next(3))
            println(m.next(5))
        }
    }
}